subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -594,6 +594,36 @@ tmpdir is currently ${System.getProperty("java.io.tmpdir")}""")
         result.assertOutputContains(string.trim())
     }
 
+    /**
+     * Given a string containing a first line which is <strong>unique within the expected output</strong>, checks
+     * if every line in the output matches the lines in the given string, reporting the failing line and number
+     * within both the actual and expected outputs.
+     *
+     * @param string the expected output
+     */
+    void outputContainsLinewise(String string) {
+        outputContainsLinewise(string.readLines())
+    }
+
+    /**
+     * Given a collection of lines beginning with a line which is <strong>unique within the expected output</strong>, checks
+     * if every line in the output matches the lines in the given string, reporting the failing line and number
+     * within both the actual and expected outputs.
+     *
+     * @param lines the expected output
+     */
+    void outputContainsLinewise(List<String> lines) {
+        List<String> outputLines = result.output.readLines()
+        def firstIndex = outputLines.findIndexOf {lines[0] == it }
+        (0..<lines.size()).each {
+            def actual = outputLines[firstIndex + it].trim()
+            def expected = lines[it].trim()
+            if (actual != expected) {
+                throw new AssertionError("Output line ${firstIndex + it}: '$actual'\ndid not match expected line $it: '$expected'")
+            }
+        }
+    }
+
```

Comment:
I don't think we want to introduce new methods like this.  The way we match output is actually more complicated if you follow what `outputContains(...)` does. We parse the output into different sections (so we can assert the right section of the output has the desired string), normalize line endings, strip out certain kinds of output and generally try to make it easier to write assertions.

Maybe we can make the failure better. It looks like we print out the output 2-3 times and the expected output 2 times.

https://github.com/gradle/gradle/blob/043f0b543fcfeb232d7dddbd014da14870e53472/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/OutputScrapingExecutionResult.java#L393-L395